ai/worker/docker.go

```diff
@@ -587,6 +599,13 @@ func (m *DockerManager) watchContainer(rc *RunnerContainer) {
 	var loadingStartTime time.Time
 	failures := 0
 	for {
+		if m.ctx.Err() != nil {
+			slog.Info("Docker manager context is done, stopping container", slog.String("container", rc.Name))
```

Comment:
This can probably be simplified to

```suggestion
			slog.Info("Docker manager context is done, stopping container", "container", rc.Name)
```